trieste/acquisition/rule.py

```diff
@@ -245,18 +245,19 @@ def remove_points(self, points_to_remove: TensorType) -> AsynchronousRuleState:
         :return: New instance of `AsynchronousRuleState` with updated pending points.
         """
 
+        @tf.function
         def _remove_point(pending_points: TensorType, point_to_remove: TensorType) -> TensorType:
-            # find indexes of all occurrences of the point to remove
-            indexes_of_point = tf.where(
-                tf.reduce_all(tf.equal(pending_points, point_to_remove), axis=1)
-            )
-            if tf.equal(tf.size(indexes_of_point), 0):
+            # find all points equal to the one we need to remove
+            are_points_equal = tf.reduce_all(tf.equal(pending_points, point_to_remove), axis=1)
```

Comment:
I'm not really familiar with the async BO code, but just a thought - would it make sense to create a dictonary for the pending points, assign them unique indices as keys and then removing them would be way easier, no?

Reply:
This code would be much simpler, yes. But that would also mean that users need to keep track of those indexes, which is really inconvenient. Tracking pending points is one of few things library can actually do for async BO without being opinionated.